sprocket-doc.md

```diff
@@ -0,0 +1,77 @@
+
+## v1.2 structs
+
+Structs defined in v1.2 WDL _will_ eventually get rich HTML documentation similar to tasks and workflows. However this has not yet been implemented, so they are given the same treatment as pre-v1.2 structs.
+
+## Parameters (inputs and outputs)
+
+Each input and output to a workflow or task should be documented, but there is some flexibility in the specifics.
+
+### Inputs
+
+Each entry in the `input` section of a task or workflow is expected to have a corresponding entry in the `parameter_meta` section. To get the most out of `sprocket doc`, it's recommended that each input entry be an object. That object should have at least a `description` key. There is special handling for the following sub-keys:
+
+- `group`: all inputs sharing the same `String` value will be rendered together in a dedicated table
+    - required inputs are _always_ rendered under the "Required Inputs" table and thus should _not_ have a `group` key (it will be ignored if present)
+    - The `Common` group of inputs will always come after the required inputs
+    - Inputs without a `group` will be rendered under "Other Inputs" which will be the last input table
+    - The `Resource` group of inputs will immediately precede the "Other Inputs" table
+    - All other groups will render alphabetically between the `Common` table and the `Resource` table.
+- `help`: will render at the top of the "Additional Meta" cell of the table
+
+If an input has a `String` value for its parameter meta entry instead of a meta object, that string value will be treated as if it were the `description` key of a meta object.
+
+### Outputs
+
+Outputs should be documented in one of two places: either in the task/workflow `meta` section under an `outputs` key or at the root of the `parameter_meta` section.
+
+Similar to inputs, each output should either be documented with an object which has a `description` key with a `String` value, or a `String` value directly. 
+
+## Meta entries with special handling
+
+All meta entries will render in the final HTML documentation, but there are some special cases we introduce.
+
+Every struct, task, and workflow `meta` section should have a `description` key with a `String` value. This description string can have Markdown formatting. The `description` string should be less than 140 characters or it will be clipped in some contexts.
+
+The `help` key should only have a `String` value, but can be of any length. It's best practice to keep `description` short, and put any additional text needed under the `help` key. Help strings can also be styled with Markdown!
+
+The `external_help` key should have a URL as its value, and will be rendered as a button which will open a new tab or window visiting the link.
+
+If a `warning` key has a `String` value, it will be rendered in a special "warning box" to draw the attention of users.
+
+Workflows can have a `category` key which will group workflow pages on the left sidebar.
```

Comment:
This definitely feels more like reference content than best-practices, and might work well as a table, e.g.,

| Key | Allowed contexts | Type | Description |
| --- | --------------- | ---- | ----------- |
| `help` | Structs, tasks, workflows | `String` | Short description (< 140 characters) that can include Markdown |

A best-practices presentation should consist primarily of an example with a link to the details for those interested.

Reply:
Oh, I think this part, whether as a table or a definition list, is great as a reference. Once there's enough fleshed out for separate reference and best practice docs, that's when the crosslinking would be beneficial.